master/internal/db/postgres_experiments.go

```diff
@@ -390,3 +394,146 @@ ORDER BY batches;`, &rows, metricName, trialID, startBatches, endBatches, startT
 	}
 	return metricSeries, endTime, nil
 }
+
+type hpImportanceDataWrapper struct {
+	TrialID int     `db:"trial_id"`
+	Hparams []byte  `db:"hparams"`
+	Batches int     `db:"batches"`
+	Metric  float64 `db:"metric"`
+}
+
+func unmarshalHPImportanceHParams(r hpImportanceDataWrapper) (model.HPImportanceTrialData, error) {
+	entry := model.HPImportanceTrialData{
+		TrialID: r.TrialID,
+		Batches: r.Batches,
+		Metric:  r.Metric,
+	}
+
+	err := json.Unmarshal(r.Hparams, &entry.Hparams)
+	if err != nil {
+		return entry, err
+	}
+	return entry, nil
+}
+
+// FetchHPImportanceTrainingData retrieves all the data needed by the hyperparameter importance
+// algorithm to measure the relative importance of various hyperparameters for one specific training
+// metric across all the trials in an experiment.
+func (db *PgDB) FetchHPImportanceTrainingData(experimentID int, metric string) (
+	[]model.HPImportanceTrialData, error) {
+	var rows []hpImportanceDataWrapper
+	var results []model.HPImportanceTrialData
+	// TODO: aren't we ignoring overtraining by taking the last?
```

Comment:
Oh I can probably remove this today - my concern was that if we rank experiments by their last metric and not their best metric, we're potentially ignoring a trial that had otherwise optimal hyperparameters, but was just overtrained. IIRC I talked to one of the machine learning engineers about this and decided that there were other reasons that's not a practical concern... I'll confirm today.

Reply:
Confirmed - this is believed to generally be a pretty good approximation for what we want, and really finding the optimal makes things more complicated. So we'll go with this and revise in the future if it proves to be a common problem.

Reply:
Cool, ok that makes sense and I agree is fine (but what do I know about ML itself lol). Just wanted to understand, thanks!